File: php/Kode program/Truck.php
<?php
require_once "Kendaraan.php";

class Truck extends Kendaraan {
    private $kapasitas;
    private $gerbong;

    public function __construct($brand, $model, $year, $price, $kapasitas, $gerbong) {
        parent::__construct($brand, $model, $year, $price);
        $this->kapasitas = $kapasitas;
        $this->gerbong = $gerbong;
    }

    public function getKapasitas() {
        return $this->kapasitas;
    }

    public function hasGerbong() {
        return $this->gerbong;
    }
}
?>

File: cpp/code program/truck.cpp
#pragma once
#include <iostream>
#include <string>
#include "kendaraan.cpp"

using namespace std;

class Truck : public Kendaraan
{
private:
    int kapasitas;
    bool gerbong;

public:
    Truck()
    {

    }

    Truck(string b, string m, int y, double p,  int c, bool ta): Kendaraan(b, m, y, p)
    {
        this->kapasitas = c;
        this->gerbong = ta;
    }

    void setkapasitas(int kapasitas)
    {
        this->kapasitas = kapasitas;
    }


    void setgerbong(bool gerbong)
    {
        this->gerbong = gerbong;
    }

    int getgerbong()
    {
        return this->gerbong;
    }


    float getkapasitas()
    {
        return this->kapasitas;
    }

    ~Truck()
    {

    }
};
